zenoh-flow/src/io/input.rs

```diff
@@ -0,0 +1,190 @@
+//
+// Copyright (c) 2022 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use crate::prelude::{ErrorKind, Message, PortId, ZFData};
+use crate::types::{Data, DataMessage, LinkMessage};
+use crate::{zferror, Result};
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::ops::Deref;
+use uhlc::Timestamp;
+
+/// The [`Inputs`](`Inputs`) structure contains all the receiving channels we created for a
+/// [`Sink`](`Sink`) or an [`Operator`](`Operator`).
+///
+/// To access these underlying channels, two methods are available:
+/// - `take`: this will return an `Input<T>` where `T` implements [`ZFData`](`ZFData`),
+/// - `take_raw`: this will return an [`InputRaw`](`InputRaw`) — a type agnostic receiver.
+///
+/// Choosing between `take` and `take_raw` is a trade-off between convenience and performance: an
+/// `Input<T>` conveniently receives instances of `T` and is thus less performant as Zenoh-Flow has
+/// to manipulate the data; an `InputRaw` is more performant but all the manipulation must be
+/// performed in the code of the Node. An `InputRaw` can be leveraged, for instance, for
+/// load-balancing or rate-limiting.
```

Comment:
I would also add that `InputRaw` is used by bindings. As Python (or other bindings in the future) cannot understand types coming from Rust (and vice-versa)